Tests/Unit/FrameworkTest.php

```diff
@@ -5346,4 +5347,34 @@ public function createBackEndUserGroupForTitleGivenStoresTitleInGroupRecord()
             )
         );
     }
+
+    /**
+     * Populate $GLOBALS['TYPO3_DB'] and create test database
+     *
+     * @see https://github.com/Nimut/testing-framework/blob/master/src/TestingFramework/TestSystem/TestSystem.php
+     *
+     * @throws Exception
+     * @throws \ReflectionException
+     * @return void
+     */
+    protected function setUpTestDatabase()
+    {
+        if (VersionNumberUtility::convertVersionNumberToInteger(TYPO3_version) >= 8001000) {
+            // The TYPO3 core misses to reset its internal connection state
+            // This means we need to reset all connections to ensure database connection can be initialized
+            $closure = \Closure::bind(function () {
+                foreach (ConnectionPool::$connections as $connection) {
+                    $connection->close();
+                }
+                ConnectionPool::$connections = [];
+            }, null, ConnectionPool::class);
+            $closure();
+
+            // get database name from $GLOBALS['TYPO3_DB']
+            $dbName = $this->getProtectedProperty($GLOBALS['TYPO3_DB'], 'databaseName');
```

Comment:
Are you sure this is working in TYPO3 v8? The database credentials structure has changed. See line 5377

Reply:
I checked this with debugging and set a breakpoint. This variable is still available. I think this extension initialize the variable somewhere ;-) I'm completely with you, this code is not nice, not robust and not safe for the future but this was not the expectation, because we don't want to use the extensions anymore.